### PR TITLE
ADEN-6827 robots.txt refactoring

### PR DESCRIPTION
**Pull request by leshniak**

### CODE REVIEW

File: extensions/wikia/RobotsTxt/classes/WikiaRobots.class.php [hunk 1]
```diff
@@ -151,11 +151,14 @@ public function configureRobotsBuilder( RobotsTxt $robots ) {
 		       $wgEnableSitemapXmlExt,
 		       $wgRobotsTxtBlockedWiki,
 		       $wgSitemapXmlExposeInRobots,
-		       $wgServer;
+			   $wgServer;
```
Comment (nandy-andy) [inline extensions/wikia/RobotsTxt/classes/WikiaRobots.class.php:154]: Something's wrong with indentation

**Review by nandy-andy — APPROVED** [review]

**Review by macbre — APPROVED** [review]

File: extensions/wikia/RobotsTxt/tests/RobotTest.php [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+			$robot->setUserAgent( $userAgent );
+		}
+		if ( !is_null( $allowPaths ) ) {
+			$robot->allowPaths( $allowPaths );
+		}
+		if ( !is_null( $disallowPaths ) ) {
+			$robot->disallowPaths( $disallowPaths );
+		}
+		$this->assertEquals( $expectedContents, $robot->getContent() );
+	}
```
Comment (mszabo-wikia) [inline extensions/wikia/RobotsTxt/tests/RobotTest.php:69]: Let's move data providers under their corresponding test method

File: extensions/wikia/RobotsTxt/tests/RobotTest.php [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+				'User-agent: *',
+				'Disallow: /',
+				''
+			],
+			$robot->getContent()
+		);
+	}
+
+	public function dataProviderClassApi() {
+		return [
```
Comment (mszabo-wikia) [inline extensions/wikia/RobotsTxt/tests/RobotTest.php:93]: You can use PHP generators (`yield`) in data providers which makes them more readable.

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]

**Comment by themech** [comment]
Closing as it got out of sync with dev

**Pull request closed without merging**